File: src/RealStatePortal.Application/Properties/PropertySearchCriteria.cs
using RealStatePortal.Domain.Enums;

namespace RealStatePortal.Application.Properties;

public sealed record PropertySearchCriteria(
    string? SearchText = null,
    PropertyType? PropertyType = null,
    string? City = null,
    decimal? PriceMin = null,
    decimal? PriceMax = null,
    int? BedroomsMin = null,
    int? BathroomsMin = null,
    decimal? AreaMin = null,
    decimal? AreaMax = null,
    PropertyStatus? Status = null,
    PropertySortOrder SortOrder = PropertySortOrder.Newest,
    int PageNumber = 1,
    int PageSize = 20);

File: frontend/src/types/PropertySearchCriteria.ts
export interface PropertySearchCriteria {
  query?: string;
  propertyType?: number;
  city?: string;
  priceMin?: number;
  priceMax?: number;
  bedroomsMin?: number;
  bathroomsMin?: number;
  areaMin?: number;
  areaMax?: number;
  sort?: "Newest" | "Oldest" | "PriceLowToHigh" | "PriceHighToLow";
}
